airbyte_cdk/sources/declarative/parsers/manifest_component_transformer.py

```diff
@@ -181,4 +194,34 @@ def propagate_types_and_parameters(
 
     @staticmethod
     def _is_json_schema_object(propagated_component: Mapping[str, Any]) -> bool:
-        return propagated_component.get("type") == "object"
+        return propagated_component.get("type") == "object" or propagated_component.get("type") == [
+            "null",
+            "object",
+        ]
+
+    @staticmethod
+    def _has_nested_components(propagated_component: Mapping[str, Any]) -> bool:
+        has_nested_components = False
+        for k, v in propagated_component.items():
+            if isinstance(v, dict) and v.get("type"):
+                has_nested_components = True
```

Comment:
This method can be simplified because once we see any dictionary that contains `type`, we can stop looping and return True immediately.

```
def _has_nested_components(propagated_component: Mapping[str, Any]) -> bool:
  for k, v in propagated_component.items():
    if isinstance(v, dict) and v.get("type"):
      return True
  return False
```